applications/GeoMechanicsApplication/tests/settlement_workflow.py

```diff
@@ -1,61 +1,195 @@
+        time = expected_stress_item["time"]
+        node_ids = [sub_item["node"] for sub_item in expected_stress_item["expected_values"]]
+        expected_nodal_stresses = [sub_item[stress_tensor_name] for sub_item in expected_stress_item["expected_values"]]
+
+        actual_nodal_stresses = test_helper.GiDOutputFileReader.nodal_values_at_time(stress_tensor_name, time, output_data, node_ids)
+
+        self.assertEqual(len(actual_nodal_stresses), len(expected_nodal_stresses))
+        for actual_total_stress, expected_total_stress in zip(actual_nodal_stresses, expected_nodal_stresses):
+            # Although the values are matrices, they are read as lists, meaning we can use assertVectorAlmostEqual
+            self.assertVectorAlmostEqual(actual_total_stress, expected_total_stress, 3)
+
+
+    def check_nodal_stresses(self):
+        reader = test_helper.GiDOutputFileReader()
+
+        for item in self.expected_stresses:
+            actual_data = reader.read_output_from(os.path.join(self.test_path, item["output_filename"]))
+            self._check_nodal_stresses_with_name(actual_data, "TOTAL_STRESS_TENSOR", item)
+            self._check_nodal_stresses_with_name(actual_data, "CAUCHY_STRESS_TENSOR", item)
+
+
+class KratosGeoMechanicsSettlementWorkflowPyRoute(KratosGeoMechanicsSettlementWorkflow):
     """
-    This test class is used to check the settlement workflow test, same as test_settlement_workflow.cpp to
-    make sure the python workflow yields the same results as the c++ workflow.
+    This test class checks the settlement workflow using the Python route.
     """
+    def get_test_dir_name(self):
+        return "python"
+
 
     def test_d_settlement_workflow(self):
-        test_name = 'test_settlement_workflow'
-        file_path = test_helper.get_file_path(test_name)
-        test_helper.run_stages(file_path, 4)
+        test_helper.run_stages(self.test_path, self.number_of_stages)
 
-        times_to_check = [1.0, 2.0, 3.0, 3.2]
+        self.check_displacements()
+        self.check_nodal_stresses()
 
-        for i in range(4):
-            result_file_name = os.path.join(file_path, f'test_model_stage{i+1}.post.res')
-            expected_result_file_name = os.path.join(file_path, f'test_model_stage{i+1}.post.orig.res')
 
-            reader = test_helper.GiDOutputFileReader()
 
-            # These node ids are at the top corner, at the bottom of the excavation,
-            # in the middle of the model and at the bottom.
-            node_ids = [1, 102, 1085, 1442]
+class KratosGeoMechanicsSettlementWorkflowCppRoute(KratosGeoMechanicsSettlementWorkflow):
+    """
+    This test class checks the settlement workflow using the C++ route.
+    """
+    def tearDown(self):
+        # The `KratosGeoSettlement` instance used by this test removes all registered GeoMechanicsApplication
+        # components when it's destroyed. If no action is taken, any following tests will start to fail. It seems
```

Comment:
Minor clarification as to why this fails the next tests
```suggestion
        # components when it's destroyed. If no action is taken, any following tests will start to fail due to components not being registered. It seems
```

Reply:
Thank you for your suggestion. I have accepted your clarification.